src/renderer/dx/DxRenderer.cpp

```diff
@@ -959,12 +958,20 @@ void DxEngine::_InvalidOr(RECT rc) noexcept
         origin.y = static_cast<float>(coord.Y * _glyphCell.cy);
 
         // Create the text layout
-        CustomTextLayout layout(_dwriteFactory.Get(),
-                                _dwriteTextAnalyzer.Get(),
-                                _dwriteTextFormat.Get(),
-                                _dwriteFontFace.Get(),
-                                clusters,
-                                _glyphCell.cx);
+        if (_drawingTextLayout == nullptr)

```

Comment:
Nothing is invalidating the `_drawingTextLayout` when the other parameters change between frames.

For instance, if `_dwriteFontFace` got updated between frames for a different font, we wouldn't recreate the layout. We would need to make sure that this is invalidated for any of the parameters that are now "cached" between frames inside the `_drawingTextLayout`.

Reply:
Yes that's what I've been thinking, too. So I left this as an idea, rather than an actual implmentation. Matter of fact, this shows as an overhead only when #2937 and #2959 is applied.

Reply:
Oh. Usually if we're talking an idea and not an actual proposal to merge, we send a draft pull request and only make it real when we're trying to actually merge it. Do you intend to advance this one after #2937 and #2959 are reviewed into an actual merged change?

Reply:
Essentially I’d to see this one merged. But #2937 and #2959 are my main focus. They have less code but more impact on overall rendering performance. So basically yes. I’d like to see #2937 and # 2959 merged before I continue working on this one.

Reply:
And let’s also take a look at #2960 which is even more harmless but still powerful